examples/ota-requestor-app/linux/LinuxOTAImageProcessor.cpp

```diff
@@ -154,12 +148,12 @@ void LinuxOTAImageProcessor::HandleProcessBlock(intptr_t context)
     if (!imageProcessor->mOfs.write(reinterpret_cast<const char *>(imageProcessor->mBlock.data()),
                                     static_cast<std::streamsize>(imageProcessor->mBlock.size())))
     {
-        downloader->OnBlockProcessed(CHIP_ERROR_WRITE_FAILED, OTADownloader::kEnd);
+        imageProcessor->mDownloader->EndDownload(CHIP_ERROR_WRITE_FAILED);
```

Comment:
Does `imageProcessor->mDownloader->EndDownload` also need to be called in `HandleFinalize` and `HandleAbort`? Or will the downloader already know it's done with download before it calls `mImageProcessor->Finalize` (and Abort) and will not need image processor to indicate so?

Reply:
Oh that's a great question. I think we need to align on a standard direction for calls like that, and I would prefer for the Downloader to call into Processor. So to answer your question - I think it should stay this way. If the Downloader is calling Finalize(), then it should have already ended the Download

Reply:
Do you agree?

Reply:
Yup, I'm good with that. Perhaps this should be clearly specified in the interface. Something to the effect that `EndDownload` should only be called if the download needs to end prematurely? Or perhaps, this should be renamed to `CancelDownload` to kind of reflect that?